### PR TITLE
Add chaos testing framework for DBFT consensus resilience validation

### PR DESCRIPTION
**Pull request by Jim8y**
## Summary

This PR introduces a comprehensive chaos testing framework to validate the resilience and fault tolerance of the DBFT consensus mechanism under various adverse network conditions and failure scenarios.

### Framework Infrastructure
- **ChaosTestBase**: Abstract base class providing common chaos test infrastructure with Akka.NET TestKit integration
- **ConsensusServiceProxy**: Actor proxy that intercepts consensus messages to inject chaos behaviors 
- **FaultInjector**: Core engine for injecting various fault conditions (message loss, corruption, byzantine behavior)
- **NetworkChaosSimulator**: Network-level chaos simulation (latency, partitions, bandwidth throttling)

### Utilities  
- **ChaosMetrics**: Comprehensive metrics collection with statistical analysis and detailed reporting
- **ChaosBenchmark**: Performance benchmarking utilities for consensus throughput measurement

## Testing Capabilities

The framework enables systematic testing of DBFT consensus under:
- **Message Loss**: Simulates network packet drops with configurable probabilities
- **Node Failures**: Tests validator failure and recovery scenarios
- **Network Partitioning**: Validates consensus behavior during network splits  
- **Byzantine Behavior**: Injects malicious validator actions
- **Performance Stress**: Measures consensus throughput under adverse conditions
- **Clock Skew**: Tests synchronization with simulated time drift

### Actor System Integration
- Proper integration with existing DBFT test infrastructure
- Uses ISigner interface for consensus service instantiation  
- Maintains compatibility with MockWallet and MockProtocolSettings
- Includes proper actor lifecycle management and cleanup

### Configuration & Reproducibility
- Environment variable-based configuration for test parameters
- Seed-based randomization for deterministic test execution
- Configurable chaos parameters (failure rates, latency ranges, etc.)
- Default values optimized for typical test scenarios

### Build System Updates
- Added Akka.TestKit dependencies to project file
- Updated package references for MSTest integration
- Maintains full compatibility with existing build pipeline

## Quality Assurance

- ✅ All existing unit tests continue to pass (904 tests in Neo.UnitTests)
- ✅ All DBFT plugin tests remain functional (34 tests) 
- ✅ Entire solution builds successfully
- ✅ Code formatted with `dotnet format`
- ✅ Proper error handling and resource cleanup implemented
- ✅ Comprehensive logging for test analysis and debugging

## Test plan

- [x] Verify all existing tests continue to pass
- [x] Confirm DBFT plugin functionality remains intact
- [x] Validate framework compiles and integrates properly
- [x] Test basic chaos injection capabilities
- [ ] Run extended chaos scenarios (future work)
- [ ] Performance benchmarking under various conditions (future work)
- [ ] Add specific test scenarios for identified edge cases (future work)

## Architecture Benefits

This framework provides:
- **Systematic Validation**: Structured approach to testing consensus resilience
- **Reproducible Testing**: Seed-based randomization ensures consistent results
- **Extensible Design**: Easy to add new chaos scenarios and metrics
- **Performance Insights**: Detailed analytics on consensus behavior under stress
- **CI/CD Integration**: Ready for integration into automated testing pipelines

The chaos testing framework establishes a solid foundation for ongoing validation of DBFT consensus robustness and can be extended with additional scenarios as the protocol evolves.

### CODE REVIEW

**Comment by Wi1l-B0t** [comment]
`Integer` & `FastInteger`.
It's confused.

I have another optimization, and maybe more clearer.

**Comment by Jim8y** [comment]
> `Integer` & `FastInteger`. It's confused.
> 
> I have another optimization, and maybe more clearer.

sure, just go ahead work on your pr. that one can keep as draft until ur pr is merged.

**Comment by Wi1l-B0t** [comment]
> > `Integer` & `FastInteger`. It's confused.
> > I have another optimization, and maybe more clearer.
> 
> sure, just go ahead work on your pr. that one can keep as draft until ur pr is merged.

Another PR, not this one

File: tests/Neo.Plugins.DBFTPlugin.Tests/ChaosTests/Framework/ConsensusServiceProxy.cs [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+using System.Threading.Tasks;
+
+namespace Neo.Plugins.DBFTPlugin.Tests.ChaosTests.Framework
+{
+    /// <summary>
+    /// Wrapper actor that intercepts messages to/from ConsensusService to inject chaos
+    /// </summary>
+    public class ConsensusServiceProxy : UntypedActor
+    {
+        private readonly IActorRef actualConsensusService;
```
Comment (Wi1l-B0t) [inline tests/Neo.Plugins.DBFTPlugin.Tests/ChaosTests/Framework/ConsensusServiceProxy.cs:32]: `actualConsensusService` -> `_actualConsensusService`.
Others as well

**Comment by Jim8y** [comment]
no longer needed, close to clean up the pr list, such that we can focus on active prs.
